py/desiutil/install.py

```diff
@@ -573,17 +573,16 @@ def set_install_dir(self):
         :class:`str`
             The directory selected for installation.
         """
-        try:
-            self.nersc = os.environ['NERSC_HOST']
-        except KeyError:
-            self.nersc = None
-        if self.options.root is None or not os.path.isdir(self.options.root):
-            if self.nersc is not None:
-                self.options.root = self.default_nersc_dir()
-            else:
-                message = "Root install directory is missing or not set."
-                self.log.critical(message)
-                raise DesiInstallException(message)
+        self.nersc = os.getenv('NERSC_HOST') #- will be None if not set
+        if self.options.root is None and self.nersc is not None:
+            self.options.root = self.default_nersc_dir()
+
+        if (self.options.root is None or not os.path.isdir(self.options.root)) \
```

Comment:
Use parenthesis to break across lines.  The backslash is ancient history.

Reply:
I have a gray beard and have been using python since before numpy existed.   But ok, parentheses it is...